### PR TITLE
add custom client

### PR DESCRIPTION
**Pull request by cleverhu**
add custom client for params and raw sql

### CODE REVIEW

**Comment by calvin0327** [comment]
@cleverhu this is very useful, we can write an example to display how to use it.

File: examples/custom-client/main.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	customClient, err := customclient.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("failed to init customClient: %v", err)
+	}
+
+	deploys := &appsv1.DeploymentList{}
+	options := builder.ListOptionsBuilder().
+		Offset(0).Limit(10).
+		RemainingCount().
+		Options()
```
Comment (Iceber) [inline examples/custom-client/main.go:29]: In the future, we can make ListOptionBuilder support methods like OptionsAndQueryParams.

**Review by Iceber — COMMENTED** [review]
LGTM

File: examples/custom-client/main.go (hunk 1 above)
Comment (calvin0327) [reply examples/custom-client/main.go:29]: yeah, we should support this approach.

**Comment by calvin0327** [comment]
LGTM

**Pull request merged**